### DISCUSSION TITLE
Extraction of individual speaker audio from overlapped speech regions

### DISCUSSION THREAD

**Discussion by PhilipAmadasun** [pyannote/pyannote-audio#1650] [Q&A]
As my title says, Is there a way for pyannote.audio to extract individual speaker audio from overlapped speech regions?

**Comment by hbredin** [comment]
Not yet but we are working on it. Stay tuned. In the meantime, you might want to have a look at Asteroid source separation toolkit.

**Reply by PhilipAmadasun** [reply]
Thanks for the reply. I had some more questions about how you guys calculate cosine similarity and of the `speaker_verification` model itself. Not sure if I should ask a whole new question as this aspect kinda also relates to this for my use case. 

* First:
        ```
        from scipy.spatial.distance import cdist
        distance = cdist(embedding1, embedding2, metric="cosine")[0,0]
        ```
        Gives an error,  I was following the tutorial [here](https://huggingface.co/pyannote/embedding)
        I have to do this instead:
        ```
        distance = cdist(np.expand_dims(embedding1,axis=0), np.expand_dims(embedding2, axis=0), metric="cosine")[0,0]
        ```
        To get results. And now I'm confused if I'm even supposed to do this. 

* Second:
       The speaker verification module gives me an error I could not find when searching through issues on the repo. Should I just create another question to ask about that?
